src/sema/functions.rs

```diff
@@ -58,7 +58,7 @@ pub fn contract_function(
                 ));
                 return None;
             }
-            if func.name.is_some() {
+            if func.name.is_some() && !ns.target.is_substrate() {
```

Comment:
This still allows constructors to have names on Substrate, without the syntax for actually using constructor names. I'm not keen on that, this is incomplete support.

Reply:
>  this is incomplete support

It is by design so we can later use the name in the metadata.